waspc/src/Wasp/Project/Db.hs

```diff
@@ -1,23 +1,65 @@
 module Wasp.Project.Db
   ( makeDevDatabaseUrl,
     databaseUrlEnvVarName,
+    getDbSystemFromPrismaSchema,
+    validateDbSystem,
+    DbSystemParseError (..),
   )
 where
 
+import Control.Exception (throwIO)
 import StrongPath (Abs, Dir, Path')
 import qualified Wasp.AppSpec as AS
-import qualified Wasp.AppSpec.App as AS.App
 import qualified Wasp.AppSpec.App.Db as AS.App.Db
+import qualified Wasp.AppSpec.App.Db as AS.Db
 import Wasp.Project.Common (WaspProjectDir)
 import qualified Wasp.Project.Db.Dev.Postgres as DevPostgres
+import qualified Wasp.Psl.Ast.ConfigBlock as Psl.ConfigBlock
+import qualified Wasp.Psl.Ast.Schema as Psl.Schema
+import Wasp.Psl.Util (findPrismaConfigBlockValueByKey)
 
-makeDevDatabaseUrl :: Path' Abs (Dir WaspProjectDir) -> [AS.Decl] -> Maybe String
-makeDevDatabaseUrl waspProjectDir decls = do
-  (appName, app) <- AS.getApp decls
-  dbSystem <- AS.App.Db.system =<< AS.App.db app
+makeDevDatabaseUrl ::
+  Path' Abs (Dir WaspProjectDir) ->
+  AS.Db.DbSystem ->
+  [AS.Decl] ->
+  Maybe String
+makeDevDatabaseUrl waspProjectDir dbSystem decls = do
+  (appName, _) <- AS.getApp decls
   case dbSystem of
-    AS.App.Db.SQLite -> Nothing
     AS.App.Db.PostgreSQL -> Just $ DevPostgres.makeDevConnectionUrl waspProjectDir appName
+    _allOtherCases -> Nothing
 
 databaseUrlEnvVarName :: String
 databaseUrlEnvVarName = "DATABASE_URL"
+
+data DbSystemParseError = UnsupportedDbSystem String | MissingDbSystem
+  deriving (Eq, Show)
+
+getDbSystemFromPrismaSchema :: Psl.Schema.Schema -> Either DbSystemParseError AS.Db.DbSystem
+getDbSystemFromPrismaSchema prismaSchema =
+  case getProviderFromPrismaSchema prismaSchema of
+    -- We parse raw config block values from Prisma file,
+    -- so we need match the provider names with quotes.
+    Just "\"postgresql\"" -> Right AS.App.Db.PostgreSQL
+    Just "\"sqlite\"" -> Right AS.App.Db.SQLite
+    Just provider -> Left $ UnsupportedDbSystem provider
+    Nothing -> Left MissingDbSystem
+
+validateDbSystem :: Either DbSystemParseError AS.Db.DbSystem -> IO AS.Db.DbSystem
```

Comment:
Ok, this new system that checks for dbSystem is quite better!

Btw I would maybe name this one `getValidDbSystemOrThrow`.

Reply:
We no longer have this function, but a great name suggestion 😄